### DISCUSSION TITLE
Docker and next.js problem

### DISCUSSION THREAD

**Discussion by clh97** [vercel/next.js#17891] [Help] [answered]
Im running a docker-compose project with two containers: app-next and app-strapi.
app-next contains a next.js application running in port 3000 (development mode)
app-strapi contains a strapi application running in port 1337

In app-next I have the following file: `pages/posts/[slug].js` and in the `getStaticPaths` function Im making a request to get the articles from strapi.

```
export async function getStaticPaths() {
  const requestUrl = getStrapiURL('/articles'); // resolves to http://app-strapi:1337/articles
  const response = await fetch(requestUrl);
  const articles = await response.json();

  return {
    paths: articles.map((article) => ({
      params: {
        slug: article.slug,
      },
    })),
    fallback: false,
  };
}
```
Inside the docker container `http://app-strapi:1337/articles` resolves correctly to my strapi instance, but in my local machine it doesn't and **that is expected**.

The problem here is: when Im loading the app-next in my local browser it loads in the first try, but if I reload the page it tries to make a XHR request to `http://app-strapi:1337/articles` from browser and I get a fetch request error.

In theory `getStaticPaths` runs at build time, right? Why this request is being made in my local browser?

To resolve this problem in my machine I aliased `app-strapi` to 127.0.0.1 in my `/etc/hosts` file.
I just want to know if there is any cleaner solution around here.

Thanks! =)

**Comment by jamesmosier** [comment, marked as answer]
Assuming this is all in development mode, [Next calls `getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#runs-on-every-request-in-development) on every request in development mode.

**Reply by clh97** [reply]
@jamesmosier thanks for your response.
Do you think adding a new field in `next.config.js` describing the desired baseUrl to call when `getStaticProps` runs in dev mode is a good idea?

**Reply by jamesmosier** [reply]
Definitely! I'd say maybe use the built-in environment variable support and put the value in a `.env` file!

https://nextjs.org/docs/basic-features/environment-variables
